File: Ohtap E Commerce Domain Models/Development/PHP/Objects/Promotions/Promotion .php
<?php
namespace Ohtap\ECommerce\DomainModels\Objects\Promotions;

use Ohtap\ECommerce\DomainModels\Objects\Discounts\IDiscountData;
use Ohtap\ECommerce\DomainModels\Objects\Files\IFileData;
use Ohtap\ECommerce\DomainModels\Objects\Packages\IPackageDataList;


class Promotion  implements IPromotion
{

    public IPackageDataList $Packages;
    public IFileData $Photo;
    public IPromotionDetailDataList $Details;
    public IProductDataList $Products;
    public string $Description;
    public int $EndDate;//long
    public int $StartDate;//long
    public IDiscountData $Discount;
    public int $State;
    public int $OrderNo;
    public string $ID;
    public string $Name;
}

File: Ohtap E Commerce Domain Models/Development/NetFramework/Models Objects/Serializable Objects/Promotions/Promotion.cs
using Ohtap.ECommerce.Domain.Models.Serializable.Objects.Products;
using Ohtap.ECommerce.Domain.Models.Serializable.Objects.PromotionDetails;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ohtap.ECommerce.Domain.Models.Objects.Files;
using Ohtap.ECommerce.Domain.Models.Objects.Packages;
using Ohtap.ECommerce.Domain.Models.Objects.Products;
using Ohtap.ECommerce.Domain.Models.Objects.PromotionDetails;
using Ohtap.ECommerce.Domain.Models.Objects.Promotions;

namespace Ohtap.ECommerce.Domain.Models.Serializable.Objects.Promotions
{
    public class Promotion : PromotionBasic, IPromotion
    {
        public IFileData Photo { get; set; }
        public IPackageDataList Packages { get; set; }

        public IProductDataList Products { get; set; }
        IPromotionDetailDataList IPromotion.Details { get; set; }
    }
}
